Add support for logging into a file or journald

jonathan-schild opened the pull request:
Supports basic log level.
Option to log and print to stdout.

Not tested under windows yet.

May crash if the user is not allowed to create/open the log file because the exception handling is not yet implemented.
However, this should not be a problem if you use default paths or paths in the user directory.

Gamer92000 commented on src/log.py:
```diff
@@ -0,0 +1,142 @@
+  AUTO = 0
+
+class LogLevel(Enum):
+  DEBUG = 0
+  INFO = 1
+  IMPORTANT = 2
```
I would prefer DEBUG, INFO, WARNING, ERROR

Gamer92000 reviewed (changes requested):
I don't see that logging to stdout will be used, as the stdout is already used for progress updates.

jonathan-schild commented:
I changed the log level and updated the methods in LogLevel.

> I don't see that logging to stdout will be used, as the stdout is already used for progress updates.
You don't have to use it xD. I thought it could be useful in debugging scenarios though.

Gamer92000 reviewed (approved):
such fancy
